### ISSUE TITLE
Type generics not getting cast when uninitialized

### ISSUE THREAD

**Issue by Ashvin-Ranjan** [nbuilding/N-lang#211]
Type generics are a bit scuffed as
```js
assert type map[str, str] = mapFrom([])
```
and
```js
assert type [a, b] (a -> b) -> list[a] -> list[b]: [x, y] (x -> y) -> list[x] -> list[y]
```
fail even when they should be cast.

**Comment by SheepTester** [comment]
This is also a bug in the JS impl

**Comment by Ashvin-Ranjan** [comment]
Closed via #213
